### PR TITLE
Introduction to Kubernetes project example

### PR DESCRIPTION
**Pull request by ricardozanini**
**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [ ] Specification
- [x] Examples
- [ ] Schema
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] Website
- [x] SDK
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:
In this PR we introduce an example project using the newly Go SDK types to create a custom Kubernetes Application using the Serverless Workflow Specification

We also added one CI check for the example itself to verify if the Kubernetes Custom Resources are being created successfully on a real cluster. We use KIND to do this check.

**Special notes for reviewers**:
Please take an extra care when reviewing the documentation

**Additional information (if needed):**

### CODE REVIEW

**Comment by ricardozanini** [comment]
Depends on #54

File: sdk/examples/kubernetes/pkg/apis/serverlessworkflow/v1alpha2/eventgroup_types.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// EventGroupSpec defines the desired state of EventGroup
+type EventGroupSpec struct {
+	// List of events deployed for this instance. An event name must be unique in the given namespace.
+	// +kubebuilder:validation:MinItems=1
+	// +listType=atomic
+	Events []sw.Eventdef `json:"events"`
+}
+
+// EventGroupStatus defines the observed state of EventGroup
+type EventGroupStatus struct {
```
Comment (n3wscott) [inline sdk/examples/kubernetes/pkg/apis/serverlessworkflow/v1alpha2/eventgroup_types.go:17]: you will want `observedGeneration` also

**Comment by tsurdilo** [comment]
@ricardozanini sorry needs a small rebase for roadmap :)

**Comment by ricardozanini** [comment]
I'll reopen this discussion on sdk-go project.

**Pull request closed without merging**